cdap-data-fabric/src/main/java/co/cask/cdap/store/DefaultNamespaceStore.java

```diff
@@ -73,9 +73,10 @@ public NamespaceMDS get() {
             table = dsCache.getDataset(APP_META_INSTANCE_ID.getId());
           } catch (DatasetInstantiationException e) {
             try {
-              table = DatasetsUtil.getOrCreateDataset(
```

Comment:
that's not what i meant, the next time call in `dsCache.getDataset(..)` would not throw exception and will be able to getDataset and also add it to the cache.